### PR TITLE
feat(pagination): create Pagination component

### PR DESCRIPTION
**Pull request by AntoninMarchardDev**
- [x] Feature
- [ ] Fix
- [ ] Enhancement

## Brief
Create a new component for Pagination.
See `README.md` file for more technical informations.

## Requirements :

:warning: Requires running `yarn` command.

## Todo - Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.

### CODE REVIEW

File: src/Pagination/index.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    const items = [];
+
+    if (pageCount <= pageRangeDisplayed) {
+      for (let i = 1; i <= pageCount; i++) {
+        items.push(getPaginationItem(i));
+      }
+    } else {
+      let left = pageRangeDisplayed / 2;
+      let right = pageRangeDisplayed - left;
+      let pageItem = index => getPaginationItem(index);
```
Comment (Artikodin) [inline src/Pagination/index.js]: Little question, why did you use `pageItem` here instead of using directly  `getPaginationItem` ?

Comment (AntoninMarchardDev) [reply src/Pagination/index.js]: @Artikodin good question, there is no valid answer so I'll remove this useless `let` :)

Comment (Artikodin) [reply src/Pagination/index.js]: cool :)

File: src/Pagination/__stories__/Pagination.stories.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    const breakValue = text('Break value', '...');
+    const alignValue = select(
+      'Alignment',
+      {
+        default: 'left',
+        center: 'center',
+        left: 'left',
+        right: 'right',
+      },
+      'default',
```
Comment (sun-tea) [inline src/Pagination/__stories__/Pagination.stories.js]: Invalid prop type on console
```
{
    center: 'center',
    left: 'left',
    right: 'right',
},
'right',
```

File: src/Pagination/__stories__/Pagination.stories.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    const pageCountValue = number(
+      'pageCount',
+      10,
+      {
+        range: true,
+        min: 1,
+        max: 1000,
+        step: 1,
+      },
+      'Page count',
```
Comment (sun-tea) [inline src/Pagination/__stories__/Pagination.stories.js]: `ALL`

File: src/Pagination/__stories__/Pagination.stories.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    const alignValue = select(
+      'Alignment',
+      {
+        default: 'left',
+        center: 'center',
+        left: 'left',
+        right: 'right',
+      },
+      'default',
+      'Alignment',
```
Comment (sun-tea) [inline src/Pagination/__stories__/Pagination.stories.js]: You can put it in `ALL`

File: src/Pagination/__stories__/Pagination.stories.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      'pageCount',
+      10,
+      {
+        range: true,
+        min: 1,
+        max: 1000,
+        step: 1,
+      },
+      'Page count',
+    );
```
Comment (sun-tea) [inline src/Pagination/__stories__/Pagination.stories.js]: You can add others props to play with them on the story
```
const pageRangeDisplayed = number(
  'pageRangeDisplayed',
  2,
  {
    range: true,
    min: 1,
    max: 5,
    step: 1,
  },
  'ALL',
);
const marginPageDisplayed = number(
  'marginPageDisplayed',
  1,
  {
    range: true,
    min: 1,
    max: 5,
    step: 1,
  },
  'ALL',
);
```

File: src/Pagination/index.js [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+  /**
+   * Render the component.
+   *
+   * @return {jsx}
+   */
+  return (
+    <Container align={align} data-testid="pagination-main-container">
+      <PaginationContainer data-testid="pagination-container">
+        <ArrowContainer side="left" onClick={handlePrevious} data-testid="arrow-left">
+          <Icon name="chevron-left" color={Theme.palette.lightGrey} />
```
Comment (sun-tea) [inline src/Pagination/index.js]: `darkGrey`

File: src/Pagination/index.js [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+        </ArrowContainer>
+        <PaginationGroup>{renderPagination()}</PaginationGroup>
+        <ArrowContainer side="right" onClick={handleNextPage} data-testid="arrow-right">
+          <Icon name="chevron-right" color={Theme.palette.lightGrey} />
```
Comment (sun-tea) [inline src/Pagination/index.js]: `darkGrey`

File: src/Pagination/elements.js [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+export const PaginationItem = styled.div`
+  border-radius: ${({ theme: { dimensions } }) => dimensions.radius};
+  background-color: ${({ theme: { palette } }) => palette.white};
+  height: 3rem;
+  width: 3rem;
+  line-heigh: 3rem;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  color: ${({ theme: { palette } }) => palette.mediumGrey};
```
Comment (sun-tea) [inline src/Pagination/elements.js]: `darkBlue`

File: src/Pagination/index.js [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+
+/**
+ * Pagination
+ *
+ * This component is in charge of displaying
+ * a Pagination
+ *
+ * @param {number} pageCount - total number of pages in the pagination.
+ * @param {number} pageRangeDisplayed - displayed page range.
+ * @param {number} marginPageDisplayed - page displayed on the pagination marges.
```
Comment (sun-tea) [inline src/Pagination/index.js]: `displayedBoundNumber`? And the doc may be clearer with something like `number of pages to display at the beginning and end of the pagination`

**Pull request closed without merging**